Update design section (#231)
AY2122S1-CS2103T-W16-2/tp: docs/DeveloperGuide.md

Here are the other classes in `Logic` (omitted from the class diagram above) that are used for parsing a user command:

<img src="images/ParserClasses.png" width="600"/>

How the parsing works:
* When called upon to parse a user command, the `AddressBookParser` class creates an `XYZCommandParser` (`XYZ` is a
  placeholder for the specific command name e.g., `AddOrderCommandParser`) which uses the other classes shown above to
  parse the user command and create a `XYZCommand` object (e.g., `AddOrderCommand`) which the `AddressBookParser`
  returns back as a `Command` object.
* All `XYZCommandParser` classes (e.g., `AddOrderCommandParser`, `DeleteOrderCommandParser`, ...) inherit from the
  `Parser` interface so that they can be treated similarly where possible e.g, during testing.

### Model component
**API** : [`Model.java`](https://github.com/AY2122S1-CS2103T-W16-2/tp/blob/master/src/main/java/ay2122s1_cs2103t_w16_2/btbb/model/Model.java)

![ModelClassDiagram](images/ModelClassDiagram.png)

The `Model` component,

* stores a `UserPref` object that represents the user’s preferences. This is exposed to the outside as a
  `ReadOnlyUserPref` objects.

> * stores the address book data in the following components:

Reviewer: ```suggestion
* stores the application data in the following components:
```

>   * a `UniqueClientList` containing `Client`s.

Reviewer: Can we change this for all 4 points?
```suggestion
  * a `UniqueClientList` containing `Client` objects.
```

  * a `UniqueIngredientList` containing `Ingredient`s.
  * a `UniqueOrderList` containing `Order`s.

>   * a `UniqueOrderList` containing `Recipe`s.

Reviewer: ```suggestion
  * a `UniqueRecipeList` containing `Recipe` objects.
```

* stores a `UserPref` object that represents the user’s preferences. This is exposed to the outside as a
  `ReadOnlyUserPref` objects.

> * stores the application in the following components:

Reviewer: ```suggestion
* stores the application data in the following components:
```

  * a `UniqueClientList` containing `Client` objects.
  * a `UniqueIngredientList` containing `Ingredient` objects.
  * a `UniqueOrderList` containing `Order` objects.

>   * a `UniqueOrderList` containing `Recipe` objects.

Reviewer: ```suggestion
  * a `UniqueRecipeList` containing `Recipe` objects.
```
